### PR TITLE
Enshrine ESDT unified syntax migration 

### PR DESCRIPTION
**Pull request by andreiblt1304**

### CODE REVIEW

File: enshrine-esdt-safe/src/to_sovereign/create_tx.rs [hunk 1]
```diff
@@ -130,7 +130,7 @@ pub trait CreateTxModule:
         refundable_payments: &ManagedVec<EsdtTokenPayment>,
     ) {
         for payment in refundable_payments {
-            self.send().direct_non_zero_esdt_payment(caller, &payment);
+            self.tx().to(caller).payment(&payment).transfer();
```
Comment (axenteoctavian) [inline enshrine-esdt-safe/src/to_sovereign/create_tx.rs:133]: You need a check if payment is non zero so you won't make transactions with 0 value. Or is the transfer function already checking this?

File: fee-market/src/subtract_fee.rs [hunk 2]
```diff
@@ -123,14 +124,17 @@ pub trait SubtractFeeModule:
...
+
+        self.tokens_for_fees()
             .insert(final_payment.fee.token_identifier.clone());
+
         self.accumulated_fees(&final_payment.fee.token_identifier)
             .update(|amt| *amt += &final_payment.fee.amount);
 
-        self.send()
-            .direct_non_zero_esdt_payment(&original_caller, &final_payment.remaining_tokens);
+        self.tx()
```
Comment (axenteoctavian) [inline fee-market/src/subtract_fee.rs:134]: You need a check if payment is non zero so you won't make transactions with 0 value. Or is the transfer function already checking this?

File: enshrine-esdt-safe/src/to_sovereign/create_tx.rs (hunk 1 above)
Comment (andreiblt1304) [reply enshrine-esdt-safe/src/to_sovereign/create_tx.rs:137]: added `require`

File: fee-market/src/subtract_fee.rs (hunk 2 above)
Comment (andreiblt1304) [reply fee-market/src/subtract_fee.rs:137]: added `require`

**Review by mihaicalinluca — APPROVED** [review]

**Pull request merged**